backend/app/api/endpoints/share.py

```
@@ -0,0 +1,539 @@
+            raise
+
+    raise HTTPException(status_code=400, detail="Invalid share token")
+
+
+@router.get(
+    "/public/knowledge",
+    summary="Get public knowledge base info",
+)
+def get_public_kb_info(
+    token: str,
+    db: Session = Depends(get_db),
```

Ruff (0.14.14)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)